src/core_editor/line_buffer.rs

```diff
@@ -454,6 +454,92 @@ impl LineBuffer {
     pub fn is_cursor_at_last_line(&self) -> bool {
         !self.get_buffer()[self.offset()..].contains('\n')
     }
+
+    /// Finds index for the first occurrence of a char to the right of offset
+    pub fn find_char_right(&self, c: &char) -> Option<usize> {
+        if self.offset() + 1 > self.lines.len() {
```

Comment:
I think this misses the case of being on the last (potentially multibyte character that can not move any further). But the search might fail already.